apps/web/src/views/CakeStaking/components/DataSet/TotalApy.tsx

```diff
@@ -0,0 +1,186 @@
+`
+
+export const TotalApy = ({ veCake, cakeAmount }: { veCake: string; cakeAmount: number }) => {
+  const { t } = useTranslation()
+  const cakePoolEmission = useCakePoolEmission()
+  const revShareEmission = useRevShareEmission()
+  const { data: totalSupply } = useVeCakeTotalSupply()
+  // CAKE Pool APR
+  const userCakeTvl = getDecimalAmount(new BigNumber(cakeAmount))
+  const userSharesPercentage = getDecimalAmount(new BigNumber(veCake)).div(totalSupply).times(100)
+
+  const cakePoolApr = useMemo(() => {
+    const apr = new BigNumber(userSharesPercentage)
+      .times(cakePoolEmission)
+      .div(3)
+      .times(24 * 60 * 60 * 365)
+      .div(userCakeTvl.div(1e18))
+      .toNumber()
+
+    return Number.isNaN(apr) ? 0 : apr
+  }, [cakePoolEmission, userCakeTvl, userSharesPercentage])
+
+  // Revenue Sharing
+  const revenueSharingApr = useMemo(() => {
+    const apr = new BigNumber(userSharesPercentage)
+      .times(revShareEmission)
+      .times(24 * 60 * 60 * 365)
+      .div(userCakeTvl)
+      .toNumber()
+
+    return Number.isNaN(apr) ? 0 : apr
+  }, [revShareEmission, userCakeTvl, userSharesPercentage])
+
+  const totalApy = useMemo(
+    () => new BigNumber(cakePoolApr).plus(revenueSharingApr).plus(BRIBE_APR).toNumber(),
+    [cakePoolApr, revenueSharingApr],
+  )
+
+  const {
+    targetRef: totalAprRef,
+    tooltip: totalAprTooltips,
+    tooltipVisible: totalAprTooltipVisible,
+  } = useTooltip(
+    <Box>
+      <Box>
+        <Text bold as="span">
+          🔹{t('Total APR')}
+        </Text>
+        <Text ml="4px" as="span">
+          {t('is the sum of veCAKE Pool APR, Revenue sharing APR and Birbes APY from Guages voting.')}
+        </Text>
+      </Box>
+      <Link mt="8px" external href="https://docs.pancakeswap.finance/products/vecake/faq#why-there-are-multiple-aprs">
+        {t('Learn More')}
+      </Link>
+    </Box>,
+    {
+      placement: 'top',
+    },
+  )
+
+  const {
+    targetRef: veCakePoolAprRef,
+    tooltip: veCakePoolAprTooltips,
+    tooltipVisible: veCakePoolAprTooltipVisible,
+  } = useTooltip(
+    <Box>
+      <Box>
+        <Text bold as="span">
+          {t('veCAKE Pool APR')}
+        </Text>
+        <Text ml="4px" as="span">
+          {t('is generated from CAKE emission, controlled by the veCAKE Pool gauge.')}
+        </Text>
+      </Box>
+      <Link mt="8px" external href="https://docs.pancakeswap.finance/products/vecake/faq#what-is-vecake-pool-apr">
+        {t('Learn More')}
+      </Link>
+    </Box>,
+    {
+      placement: 'top',
+    },
+  )
+
+  const {
+    targetRef: revenueSharingPoolAprRef,
+    tooltip: revenueSharingPoolAprTooltips,
+    tooltipVisible: revenueSharingPoolAprTooltipVisible,
+  } = useTooltip(
+    <Box>
+      <Box>
+        <Text bold as="span">
+          {t('Revenue Sharing APR')}
+        </Text>
+        <Text ml="4px" as="span">
+          {t('is generated from weekly revenue sharing from PancakeSwap V3.')}
+        </Text>
```

Comment:
Just for this

Reply:
i was thinking more of creating the component in the same file, not as a exportale component. i just thought it would be nice for reduciing jsx code. but was just a suggestion. 

```ts
  // maybe define config in utils?? or same file
  const TooltipType = {
    TOTAL_APR: 'TOTAL_APR',
    VECAKE_POOL_APR: 'VECAKE_POOL_APR',
    REVENUE_SHARING_APR: 'REVENUE_SHARING_APR',
    ...
  };
  
  // Function to create a tooltip instance based on the tooltip type
  // Configuration object for tooltips
  const tooltipConfigs = {
    [TooltipType.TOTAL_APR]: {
      title: '🔹' + t('Total APR'),
      description: t('is the sum of veCAKE Pool APR, Revenue sharing APR and Birbes APY from Guages voting.'),
      href: 'https://docs.pancakeswap.finance/products/vecake/faq#why-there-are-multiple-aprs',
      placement: 'top'
    },
    [TooltipType.VECAKE_POOL_APR]: {
      title: t('veCAKE Pool APR'),
      description: t('is generated from CAKE emission, controlled by the veCAKE Pool gauge.'),
      href: 'https://docs.pancakeswap.finance/products/vecake/faq#what-is-vecake-pool-apr',
      placement: 'top'
    },
    [TooltipType.REVENUE_SHARING_APR]: {
      title: t('Revenue Sharing APR'),
      description: t('is generated from weekly revenue sharing from PancakeSwap V3.'),
      href: 'https://docs.pancakeswap.finance/products/vecake/faq#what-is-revenue-sharing-apr',
      placement: 'top'
    }
  };

  const createTooltipInstance = (type) => {
    const config = tooltipConfigs[type];
    const {
      targetRef,
      tooltip,
      tooltipVisible,
    } = useTooltip(
      <Box>
        <Box>
          <Text bold as="span">
            {config.title}
          </Text>
          <Text ml="4px" as="span">
            {config.description}
          </Text>
        </Box>
        <Link mt="8px" external href={config.href}>
          {t('Learn More')}
        </Link>
      </Box>,
      {
        placement: placement,
      },
    );
  
    return {
      targetRef,
      tooltip,
      tooltipVisible,
    };
  
  };
  

// Create individual tooltip instances
const totalAprTooltip = createTooltipInstance(TooltipType.TOTAL_APR);
const veCakePoolAprTooltip = createTooltipInstance(TooltipType.VECAKE_POOL_APR);
const revenueSharingAprTooltip = createTooltipInstance(TooltipType.REVENUE_SHARING_APR);

return (
  <Flex width="100%" flexDirection="column">
    <Flex justifyContent="space-between">
      <TooltipText fontSize="14px" color="textSubtle" ref={totalAprRef}>
        {t('Total APR')}
      </TooltipText>
      <Flex>
        <Text>🔹</Text>
        <GradientText>{t('Up to %apr%%', { apr: totalApy.toFixed(2) })} </GradientText>
      </Flex>
    </Flex>
    <Box ml="25px">
      <Flex mt="4px" justifyContent="space-between">
        <TooltipText fontSize="14px" color="textSubtle" ref={veCakePoolAprRef}>
          {t('veCAKE Pool APR')}
        </TooltipText>
        <Text>{`${cakePoolApr.toFixed(2)}%`}</Text>
      </Flex>
      <Flex mt="4px" justifyContent="space-between">
        <TooltipText fontSize="14px" color="textSubtle" ref={revenueSharingPoolAprRef}>
          {t('Revenue Sharing APR')}
        </TooltipText>
        <Text>{`${revenueSharingApr.toFixed(2)}%`}</Text>
      </Flex>
      <Flex mt="4px" justifyContent="space-between">
        <TooltipText fontSize="14px" color="textSubtle" ref={bribeAprRef}>
          {t('Bribe APR')}
        </TooltipText>
        <GradientText>{`${BRIBE_APR.toFixed(2)}%`}</GradientText>
      </Flex>
    </Box>
    {totalAprTooltip}
    {veCakePoolAprTooltip}
    {revenueSharingPoolAprTooltipVisible && revenueSharingPoolAprTooltips}
    {revenueSharingAprTooltip}
  </Flex>
)

```